### PR TITLE
North angle

### LINKED ISSUE #214: North Axis not synchronized between OSM and Floorspace JSON
App seems to be using old `config["initialNorthAxis"]` key to store north angle:

https://github.com/openstudiocoalition/OpenStudioApplication/blame/develop/src/openstudio_lib/GeometryEditorView.cpp#L273

Correct key is `project["north_axis"]`:

https://github.com/NREL/OpenStudio/blob/develop/src/utilities/geometry/FloorplanJS.cpp#L1508
https://github.com/NREL/floorspace.js/blame/develop/schema/geometry_schema.json#L122

### PR DESCRIPTION
**Pull request by macumber**
Fixes #214

Depends on https://github.com/NREL/OpenStudio/issues/4054

### CODE REVIEW

**Comment by jmarrec** [comment]
Obviously since CI uses OS 3.0.1 it will never build it correctly.

File: src/openstudio_lib/GeometryEditorView.cpp [hunk 1]
```diff
@@ -196,6 +208,13 @@ FloorspaceEditor::FloorspaceEditor(const openstudio::path& floorplanPath, bool i
       model::FloorplanJSForwardTranslator ft;
       m_floorplan = ft.updateFloorplanJS(*m_floorplan, m_model, true);
 
+      // update north axis from model, floorspace js northAxis is opposite of EnergyPlus's
+      if (building) {
+        m_floorplan->setNorthAxis(-building->northAxis());
+      }
```
Comment (jmarrec) [inline src/openstudio_lib/GeometryEditorView.cpp:214]: Hum...

File: src/openstudio_lib/GeometryEditorView.cpp [hunk 2]
```diff
@@ -487,6 +519,14 @@ void FloorspaceEditor::translateExport() {
   if (model) {
     m_exportModel = *model;
     m_exportModelHandleMapping = rt.handleMapping();
+
+    // manually add mappings between current model and new model for Site, Facility, and Building objects
+    m_exportModelHandleMapping[m_model.getUniqueModelObject<model::Site>().handle()] = m_exportModel.getUniqueModelObject<model::Site>().handle();
+    m_exportModelHandleMapping[m_model.getUniqueModelObject<model::Facility>().handle()] =
+      m_exportModel.getUniqueModelObject<model::Facility>().handle();
+    m_exportModelHandleMapping[m_model.getUniqueModelObject<model::Building>().handle()] =
+      m_exportModel.getUniqueModelObject<model::Building>().handle();
```
Comment (jmarrec) [inline src/openstudio_lib/GeometryEditorView.cpp:528]: Why is this needed? I'm not doubting it isn't, just curious.

**Review by jmarrec — APPROVED** [review]
As long as it builds and tests fine, I'm ok with it.

File: src/openstudio_lib/GeometryEditorView.cpp (hunk 2 above)
Comment (macumber) [reply src/openstudio_lib/GeometryEditorView.cpp:528]: This one is pretty weird.  Basically we don't have site, facility, or building objects in the FloorspaceJS file so we don't have anywhere to store the corresponding OpenStudio Model handles.

File: src/openstudio_lib/GeometryEditorView.cpp (hunk 1 above)
Comment (macumber) [reply src/openstudio_lib/GeometryEditorView.cpp:214]: What was the concern, the difference in north angle convention?

**Comment by macumber** [comment]
Thanks for the review @jmarrec, it does build provided we hook it up with the current OS SDK package.  NREL merged https://github.com/NREL/OpenStudio/issues/4054 so we should be able to update to use the packages on that PR. I'm not 100% sure how to do that since my work on https://github.com/openstudiocoalition/OpenStudioApplication/pull/208 was incorrect.

File: src/openstudio_lib/GeometryEditorView.cpp (hunk 2 above)
Comment (macumber) [reply src/openstudio_lib/GeometryEditorView.cpp:528]: I did leave some comments in the OS SDK: https://github.com/macumber/OpenStudio/blob/ac01edda9d24618b9c552f623e0a5c92bdd5de86/src/model/ThreeJSReverseTranslator.hpp#L67

**Comment by macumber** [comment]
Blocked by https://github.com/openstudiocoalition/OpenStudioApplication/issues/225

File: src/openstudio_lib/GeometryEditorView.cpp (hunk 1 above)
Comment (jmarrec) [reply src/openstudio_lib/GeometryEditorView.cpp:214]: Yeah.

**Comment by jmarrec** [comment]
@macumber After #208, I rebased onto develop (which pulls OpenStudio 3.1.0-alpha) and force pushed, CI should now build it correctly.

**Comment by jmarrec** [comment]
@macumber this seems fine and CI is fine with it too. I wonder if we couldn't benefit from tests though?

**Comment by macumber** [comment]
Very cool, thanks @jmarrec!
